scripts/core/memory.yml

```diff
@@ -540,3 +540,99 @@ type: class
 desc: "C++ wrapper for memory allocation"
 name: $xMem
 owner: $xContext
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "atomic access attribute flags"
+class: $xMem
+name: $x_memory_atomic_attr_exp_flags_t
+version: "1.7"
+etors:
+    - name: NO_ATOMICS
+      desc: "Atomics on the pointer are not allowed"
+    - name: NO_HOST_ATOMICS
+      desc: "Host atomics on the pointer are not allowed"
+    - name: HOST_ATOMICS
+      desc: "Host atomics on the pointer are allowed. Requires $X_MEMORY_ACCESS_CAP_FLAG_ATOMIC returned by $xDeviceGetMemoryAccessProperties."
+    - name: NO_DEVICE_ATOMICS
+      desc: "Device atomics on the pointer are not allowed"
+    - name: DEVICE_ATOMICS
+      desc: "Device atomics on the pointer are allowed. Requires $X_MEMORY_ACCESS_CAP_FLAG_ATOMIC returned by $xDeviceGetMemoryAccessProperties."
+    - name: NO_SYSTEM_ATOMICS
+      desc: "Concurrent atomics on the pointer from both host and device are not allowed"
+    - name: SYSTEM_ATOMICS
+      desc: "Concurrent atomics on the pointer from both host and device are allowed. Requires $X_MEMORY_ACCESS_CAP_FLAG_CONCURRENT_ATOMIC returned by $xDeviceGetMemoryAccessProperties."
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Sets atomic access attributes for a shared allocation"
+class: $xMem
+name: SetAtomicAccessAttributeExp
+version: "1.7"
+details:
+    - |
+      If the shared-allocation is owned by multiple devices (i.e. nullptr
```

Comment:
how this API works if it is called multiple times with different flags?
is latest call replacing all previous calls or is it additive?

Reply:
I vote for replacing previous setting. Makes driver simpler.

Reply:
@MichalMrozek : please see last commit